eudemo/eudemo/maintenance/equatorial_port.py

```diff
@@ -0,0 +1,229 @@
+from bluemira.geometry.solid import BluemiraSolid, BluemiraWire
+from bluemira.geometry.tools import (
+    boolean_fuse,
+    extrude_shape,
+    make_polygon,
+    offset_wire,
+    slice_shape,
+)
+
+
+class EquatorialPort(ComponentManager):
+    """
+    Wrapper around a Equatorial Port component tree
+    """
+
+    def xz_boundary(self) -> BluemiraWire:
+        """Returns a wire defining the x-z boundary of the Equatorial Port"""
+        return (
+            self.component.get_component("xz")
+            .get_component(CastellationBuilder.NAME)
+            .shape.boundary[0]
+        )
+
+
+@dataclass
+class EquatorialPortDesignerParams(ParameterFrame):
+    """
+    Equatorial Port Designer parameters
+    """
+
+    ep_height: Parameter[float]
+
+
+@dataclass
+class CastellationBuilderParams(ParameterFrame):
+    """
+    Castellation Builder parameters
+    """
+
+    n_components: Parameter[int]
+    cst_r_corner: Parameter[float]
+
+
+class EquatorialPortDesigner(Designer):
+    """
+    Equatorial Port Designer
+    """
+
+    param_cls: Type[EquatorialPortDesignerParams] = EquatorialPortDesignerParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, EquatorialPortDesignerParams],
+        build_config: Union[Dict, None],
+        koz_z_offset: float,
+        x_ib: float,
+        x_ob: float,
+    ):
+        super().__init__(params, build_config)
+        self.koz_offset = koz_z_offset
+        self.x_ib = x_ib
+        self.x_ob = x_ob
+
+    def run(self) -> BluemiraWire:
+        """
+        Design the xz keep-out zone profile of the equatorial port
+        """
+        x_ib = self.x_ib
+        x_ob = self.x_ob
+        z_h = self.params.ep_height.value + (2 * self.koz_offset)
+
+        x = (x_ib, x_ob, x_ob, x_ib)
+        z = (-z_h / 2, -z_h / 2, z_h / 2, z_h / 2)
+
+        ep_boundary = BluemiraFace(
+            make_polygon({"x": x, "y": 0, "z": z}, label="koz", closed=True)
+        )
+        return ep_boundary
+
+
+class CastellationBuilder(Builder):
+    """
+    Castellation Builder
+    """
+
+    NAME = "Castellation"
+    param_cls: Type[CastellationBuilderParams] = CastellationBuilderParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, CastellationBuilderParams],
+        build_config: Union[Dict, None],
+        length: float,
+        start_profile: BluemiraFace,
+        extrude_vec: tuple,
+        offsets: list,
+        castellation_offsets: list,
+    ):
+        super().__init__(params, build_config)
+        self.length = length
+        self.face = start_profile
+        self.vec = np.array(extrude_vec)
+        self.off = offsets
+        self.cst = castellation_offsets
+
+    def build(self) -> Component:
+        """
+        Build the castellated component.
+        """
+        # TODO: Implement corner radii using PR #1992
+        self.r_rad = self.params.cst_r_corner.value
```

Comment:
Note to future self: it is correct to have this parameter in this builder.

Explanation: I was originally thinking that you don't need this, seeing as the radii could be built into the profile input, however, if we want the radii to be same on all offset shapes (likely), then offsetting outwards would increase the radius.

Might be worth specifying in the builder docstring that the radii are handled by the builder.

Reply:
May be worth pre-empting this by labelling or listing wires that require fillet/chamfer operations. 
It's not every wire that needs this (not the inboard and outboard face boundaries), so would like to ask if it's wires defining internal and external corners, or just external corners that would go into the list